core/src/main/java/software/amazon/awssdk/core/client/builder/ClientAsyncHttpConfiguration.java

```diff
@@ -68,7 +68,7 @@ private ClientAsyncHttpConfiguration(DefaultHttpConfigurationBuilder builder) {
      * to ease resolution of the client. Returns an empty {@link Optional} if neither is set.
      */
     @SdkInternalApi
-    Optional<Either<SdkAsyncHttpClient, SdkAsyncHttpClientFactory>> toEither() {
+    public Optional<Either<SdkAsyncHttpClient, SdkAsyncHttpClientFactory>> toEither() {
```

Comment:
Hmm I don't like this. Is there anything we can do about it? Maybe a ReviewBeforeRelease annotation as I believe we want to flatten this out anyways which could make this cleaner.